metrics/gaugeReporter.go

```diff
@@ -0,0 +1,26 @@
+package metrics
+
```

Comment:
This is a nice refactoring.

However, the naming of this file could be better and not a gaugeReporter. Secondly it seems generic that it could be reused for any caller functions. If this file needs to deal with only metric reporters then it needs to be refactored accordingly.
For ex. This file simply registers a callback function and not necessarily validates that the registered function reports only metrics. In future I could add register any function (logging, closing connections, anything...) that gets called inside the goroutine.